src/main.rs

```diff
@@ -18,6 +18,43 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+use clap::Parser;
+use std::fs::{write, File};
+use std::io::BufReader;
+use std::path::Path;
+use std::process::exit;
+
+mod disasm;
+
+use disasm::Disasm;
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct Cli {
+    #[arg(short, long, value_parser = file_exists)]
+    input: String,
+}
+
+fn file_exists(input: &str) -> Result<String, String> {
```

Comment:
`file_exists` or in general a function that it can ask question about the input argument should be return a boolean, so call this function something like `file_read` or `file_parse`.

Reply:
Che ne pensi di `file_check`?

Reply:
Oppure `path_parse`?